src/CSET/operators/plot.py

```diff
@@ -259,57 +268,74 @@ def _colorbar_map_levels(cube: iris.cube.Cube):
     for varname in varnames:
         # Get the colormap for this variable.
         try:
-            cmap = mpl.colormaps[colorbar[varname]["cmap"]]
-            if pressure_level is None:
-                var_colorbar = colorbar[varname]
-            else:
-                # If pressure level is specified for cube use a pressure-level
-                # specific colorbar, if one exists.
-                try:
-                    var_colorbar = colorbar[varname]["pressure_levels"][pressure_level]
-                except KeyError:
-                    logging.warning(
-                        "%s has no colorbar definition for pressure level %s.",
-                        varname,
-                        pressure_level,
-                    )
-                    # Fallback to variable default.
-                    var_colorbar = colorbar[varname]
+            var_colorbar = colorbar[varname]
+            cmap = plt.get_cmap(colorbar[varname]["cmap"], 51)
+            varname_key = varname
```

Comment:
```suggestion
            varname_key = varname
            break
```
We should stop looping after we have found a colourbar, as otherwise we will end up using the less specific one.